src/WebJobs.Script.WebHost/WebScriptHostManager.cs

```diff
@@ -394,9 +420,39 @@ protected override void OnHostStarted()
                 _secretManager.PurgeOldSecretsAsync(Instance.ScriptConfig.RootScriptPath, Instance.TraceWriter, Instance.Logger);
             }
 
+            if (Microsoft.Azure.WebJobs.Script.WebHost.Management.FunctionsSyncManager.IsSyncTriggersEnvironment(_settingsManager))
```

Comment:
In v2 this is done in a hosted service https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script.WebHost/FunctionsSyncService.cs.